sklearn/cluster/_birch.py

```diff
@@ -147,7 +149,7 @@ class _CFNode:
 
     """
 
-    def __init__(self, *, threshold, branching_factor, is_leaf, n_features):
+    def __init__(self, *, threshold, branching_factor, is_leaf, n_features, dtype=None):
```

Comment:
We can pass it all the time instead of having a default:

```suggestion
    def __init__(self, *, threshold, branching_factor, is_leaf, n_features, dtype):
```